tools/resolve_manifest_file_test.go

```diff
@@ -0,0 +1,109 @@
+package tools
+
+import (
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"os"
+	"path/filepath"
+	"testing"
+)
+
+//{
+//	{"input, valid manifest", true, false, "substreams.yaml", "substreams.yaml", false},
+//	{"input, invalid manifest", false, false, "substreams.yaml", "", true},
+//	{"input, valid dir", true, true, "manifests", "manifest/substreams.yaml", false},
+//	{"input, invalid dir", false, false, "manifests", "", true},
+//	{"no input, has manifest", true, false, "", "substreams.yaml", false},
+//	{"no input, no manifest", false, false, "", "", true},
+//}
+
+func TestResolveManifestFile(t *testing.T) {
+	type args struct {
+		input        string
+		subDirectory string
+		filesOnDisk  []string
+	}
+	tests := []struct {
+		name             string
+		args             args
+		wantManifestName string
+		assertion        require.ErrorAssertionFunc
+	}{
+		{
+			"no input provided",
+			args{"", "", []string{"substreams.yaml"}},
+			"substreams.yaml",
+			require.NoError,
+		},
+		{
+			"no input provided and not substreams.yaml present",
+			args{"", "", []string{}},
+			"",
+			errorEqual("no manifest entered in dir w/o a manifest"),
+		},
+		{
+			"input provided, valid manifest file",
+			args{"substreams-custom.yaml", "", []string{"substreams-custom.yaml"}},
+			"substreams-custom.yaml",
+			require.NoError,
+		},
+		{
+			"input provided, invalid manifest file",
+			args{"substreams-custom.yaml", "", []string{}},
+			"",
+			errorEqual("reading input: stat substreams-custom.yaml: no such file or directory"),
+		},
+		{
+			"input provided, valid dir",
+			args{"manifests-dir", "manifests-dir", []string{"substreams.yaml"}},
```

Comment:
`"manifests-dir", "manifests-dir"` makes it hard to reason about args, `"manifests-dir", "sub-dir"` helps a bit

Reply:
thats not the intended purpose, but highlights the problem that my variable name is not as accurate as it should be. Fixed

Reply:
Ok good